src/Gitee/Provider.php

```diff
@@ -78,7 +78,7 @@ protected function getEmailByToken($token)
         }
 
         foreach (json_decode($response->getBody(), true) as $email) {
-            if ($email['state'] == 'confirmed' && in_array('primary', $email['scope'])) {
+            if ($email['state'] === 'confirmed' && in_array('primary', $email['scope'])) {
```

Comment:
Maybe worth adding `true` for strict in array comparison as well?

Reply:
I don't think so, strict comparison with `===` is about the type, as `$strict` argument is about the case.

Reply:
Sorry, not sure what you mean here? Below will use strict comparison instead of loose:

> If the third parameter strict is set to TRUE then the in_array() function will also check the types of the needle in the haystack.

```suggestion
            if ($email['state'] === 'confirmed' && in_array('primary', $email['scope'], true)) {
```

Reply:
Oh sorry, I didn't get you before either.
I've also added the strict comparison when using `is_array` function. See 2fb83e2909329e147f9ab0ca942b8b3751f10d48